VideoLocker/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -138,12 +143,71 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
             @Override
             public void onClick(View v) {
-                startActivityForResult(helper.createChooserIntent(getActivity()), CHOOSE_PHOTO_REQUEST);
+                final PopupMenu popup = new PopupMenu(getActivity(), v);
+                popup.getMenuInflater().inflate(R.menu.change_photo, popup.getMenu());
+                popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+                    public boolean onMenuItemClick(MenuItem item) {
+                        switch (item.getItemId()) {
+                            case R.id.take_photo: {
+                                startActivityForResult(
+                                        helper.createCaptureIntent(getActivity()),
+                                        CAPTURE_PHOTO_REQUEST);
+                                break;
+                            }
+                            case R.id.choose_photo: {
+                                final Intent galleryIntent = new Intent()
+                                        .setType("image/*")
+                                        .setAction(Intent.ACTION_GET_CONTENT);
+                                startActivityForResult(galleryIntent, CHOOSE_PHOTO_REQUEST);
+                                break;
+                            }
+                            case R.id.remove_photo: {
+                                executePhotoTask(new DeleteAccountImageTask(getActivity(), username) {
+                                    @Override
+                                    protected void onSuccess(Void aVoid) throws Exception {
+                                        hideLoading();
+                                    }
+
+                                    @Override
+                                    protected void onException(Exception e) throws RuntimeException {
+                                        super.onException(e);
+                                        showErrorMessage(e);
+                                    }
+
+                                    private void hideLoading() {
+                                        if (null != viewHolder) {
+                                            viewHolder.profileImageProgress.clearAnimation();
+                                            viewHolder.profileImageProgress.setVisibility(View.GONE);
+                                        }
+                                    }
+                                });
+                                break;
+                            }
+                        }
+                        return true;
+                    }
+                });
+                popup.show();
             }
         });
         setData(account, formDescription);
     }
 
+    private void executePhotoTask(Task task) {
+        viewHolder.profileImageProgress.setVisibility(View.VISIBLE);
+        if (viewHolder.profileImageProgress.getAnimation() == null) {
```

Comment:
The `IconView` now has built-in support for rotation animation. We should migrate to that now.